carbon-design-system/carbon
# fix: slider to range step values

Closes #14455 

Javascript floating point maths is a bit problematic.

0.3 + 0.3 + 0.3 evaluates to 0.89999999999'

The range control does not normally have a problem with this, ensuring the nearest fixed length value is used.

While this could be fixed by replicating the math inside the standard input control, the simplest solution is to set the value of a range control and read it back. This PR does just that.

#### Changelog

**Changed**

- packages/react/src/components/Slider/Slider.tsx

#### Testing / Reviewing

1. Run Storybook locally and selecting the `Slider` component.
2. Set `Step` to 0.3.
3. Drag the handle to the value 0.
4. Press the up arrow 3 times.

The result should be 0.9 but is 0.8999999' due to JS floating point math.

Applying this PR we can see the value is 0.9.
